### PR TITLE
FE-71: Add link heading with icon on hover at the end.

### PR DESCRIPTION
**Pull request by zhengmar3**
### Overview

"It would be more intuitive to also have the little ‘link' icon can display beside each heading to make that more intuitive. Probably light grey (so as not in the reader’s face) to the right of each heading. This means that when the reader clicks the link beside a header, the URL at the top changes from [www.xyz.com](http://www.xyz.com/) to [www.xyz.com#headinglink](http://www.xyz.com/#headinglink). That enables customer support and users to send links directly to relevant content, which is helpful as some of our pages are rather large. Probably only relevant for H2"

JIRA: https://r3-cev.atlassian.net/browse/FE-71

Based on the ticket requirement. All headings from `H2` downward have been changed to link heading, with the 🔗 unicode icon at the end only visible when on hovering.

### Screenshots

`Application Networks` heading when on hover state:

Screenshot 2023-04-24 at 11 45 12

When user click on the link icon. the page position will change to highlight the heading from the top. Also the URL change to 
`/key-concepts.html#application-networks`

Screenshot 2023-04-24 at 11 47 27

### CODE REVIEW

**Review by yarivus — APPROVED** [review]
LGTM - thanks for the detailed explanations

File: themes/doks/layouts/_default/_markup/render-heading.html
Comment (zhengmar3) [reply themes/doks/layouts/_default/_markup/render-heading.html:1]: The icon is an unicode so bascially a text, no svg, png file needed. And better to style than plain .png file provided
Also it's fastest to render. 

I also put an opacity to 0.6, it's light grey if you look at the screenshot?

Comment (conor-r3) [inline themes/doks/layouts/_default/_markup/render-heading.html:1]: Is it possible to break this across 3 lines to indent the anchor inside the header and make it easier to read?

Comment (zhengmar3) [reply themes/doks/layouts/_default/_markup/render-heading.html:1]: Another reason not to use the white-background .png icon provided is that it will not work with dark mode.

Comment (conor-r3) [reply themes/doks/layouts/_default/_markup/render-heading.html:1]: If the docs team are happy with it then okay, but please wait for them to approve before merging.

**Review by yarivus — APPROVED** [review]
LGTM

**Comment by nadinequinr3** [comment]
I've built a preview here: https://nadine.preview.docs.r3.com/

**Review by nadinequinr3 — APPROVED** [review]

**Pull request merged**
